acceptance/tests/proxy_connection.rb

```diff
@@ -0,0 +1,120 @@
+require 'pxp-agent/test_helper.rb'
+require 'pxp-agent/task_helper.rb'
+require 'puppet/acceptance/environment_utils.rb'
+
+test_name 'Connect via proxy' do
+  extend Puppet::Acceptance::EnvironmentUtils
+
+  # init
+  squid_log = "/var/log/squid/access.log"
+  setup_squid_proxy(master)
+  proxy = "http://#{master}:3128"
```

Comment:
If this is going to be hard-coded, we should have `setup_squid_proxy` return the proxy address.